packages/common/src/services/auth/hedgehog.ts

```diff
@@ -175,5 +185,42 @@ export const createHedgehog = ({
     }
   }
 
+  const originalConfirmCredentials = hedgehog.confirmCredentials.bind(hedgehog)
+  hedgehog.confirmCredentials = async ({
```

Comment:
It feels kinda weird that we're mutating these fields. but I guess we do it for the other functions too. Maybe theres a different way though? although passing in the constructor doesn't make sense, and also can't use spread bc hedgehog is a class so idk

Reply:
Agreed. I was following the existing pattern here, but this whole class could stand to be cleaned up after we're done removing libs.